Source/monster.cpp

```diff
@@ -1208,15 +1208,15 @@ void MonsterAttackPlayer(Monster &monster, Player &player, int hit, int minDam,
 	if (HasAnyOf(player._pIFlags, ItemSpecialEffect::Thorns) && monster.mode != MonsterMode::Death) {
 		int mdam = (GenerateRnd(3) + 1) << 6;
 		ApplyMonsterDamage(DamageType::Physical, monster, mdam);
-		if (monster.hitPoints >> 6 <= 0)
+		if (!monster.IsAlive())
 			M_StartKill(monster, player);
 		else
 			M_StartHit(monster, player, mdam);

```

Comment:
Question should we change the condition to avoid the negation?
(Same for other places)

```suggestion
		if (monster.IsAlive())
			M_StartHit(monster, player, mdam);
		else
			M_StartKill(monster, player);
```